src/main/kotlin/com/sourcegraph/cody/chat/ui/SingleMessagePanel.kt

```diff
@@ -28,23 +28,38 @@ class SingleMessagePanel(
     private val chatSession: ChatSession,
 ) : PanelWithGradientBorder(gradientWidth, chatMessage.speaker) {
   private var lastMessagePart: MessagePart? = null
+  private var lastTrimmedText = ""
 
   init {
     val markdownNodes: Node = markdownParser.parse(chatMessage.actualMessage())
     markdownNodes.accept(MessageContentCreatorFromMarkdownNodes(this, htmlRenderer))
   }
 
-  fun updateContentWith(message: ChatMessage) {
-    val markdownNodes = markdownParser.parse(message.actualMessage())
-    val lastMarkdownNode = markdownNodes.lastChild
-    if (lastMarkdownNode != null && lastMarkdownNode.isCodeBlock()) {
-      val (code, language) = lastMarkdownNode.extractCodeAndLanguage()
-      addOrUpdateCode(code, language)
-    } else {
-      val nodesAfterLastCodeBlock = markdownNodes.findNodeAfterLastCodeBlock()
-      val renderedHtml = htmlRenderer.render(nodesAfterLastCodeBlock)
-      addOrUpdateText(renderedHtml)
+  fun updateContentWith(text: String) {
```

Comment:
nit: It would be nice to have a comment over this method. The logic seems to be more complex than needed (I know it's needed today but tomorrow... 😅 )